connectors/protocol/connectors.py

```diff
@@ -179,18 +183,23 @@ async def connector_put(
             is_native=is_native,
         )
 
-    async def connector_exists(self, connector_id):
+    async def connector_exists(self, connector_id, include_deleted=False):
         try:
-            doc = await self.fetch_by_id(connector_id)
+            doc = await self.get_connector(connector_id, include_deleted)
             return doc is not None
-        except DocumentNotFoundError:
+        except ElasticNotFoundError:
```

Comment:
What's the difference with `DocumentNotFoundError`?

Reply:
`ElasticNotFoundError` is a `NotFound` error from `elasticsearch` package. We do `from elasticsearch import ( NotFoundError as ElasticNotFoundError)` to make it explicit that it comes from es client.

Framework implements it's own [DocumentNotFoundError](https://github.com/elastic/connectors/blob/b1d4f221a6afa1c7eb1095d451dfeb015f4975a4/connectors/es/index.py#L16) but IMO we should stay consistent with ES client error defs, wdyt? Otherwise I would need to catch not found just to use our own error class. They are not compatible.

 For now it's just used for this endpoint